addOns/client/src/main/java/org/zaproxy/addon/client/ExtensionClientIntegration.java

```diff
@@ -623,11 +650,14 @@ private ZapMenuItem getMenuItemCustomScan() {
 
     public ImageIcon getIcon() {
         if (icon == null) {
-            icon =
-                    DisplayUtils.getScaledIcon(
-                            ExtensionClientIntegration.class.getResource(
-                                    ExtensionClientIntegration.RESOURCES + "/spiderClient.png"));
+            icon = getIcon("spiderClient.png");
         }
         return icon;
     }
+
+    public static ImageIcon getIcon(String name) {
+        return DisplayUtils.getScaledIcon(
+                ExtensionClientIntegration.class.getResource(
```

Comment:
It could log an error when the URL is null, it would make it easier to catch incorrect paths.